### PR TITLE
Ansible Tower user and credential module

### PR DESCRIPTION
**Pull request by wwitzel3**
##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME

Ansible Tower user module
Ansible Tower credential module

##### ANSIBLE VERSION

```
ansible 2.3.0 (tower_module 57b3783869) last updated 2017/02/03 13:47:23 (GMT -400)
  config file =
  configured module search path = Default w/o overrides
```

##### SUMMARY

Allows playbook authors to create Ansible Tower users and credentials. There will be a followup to this PR to add more functionality.

Initial module was completed and merged in #20355.

### CODE REVIEW

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    defaults to find your Tower host information.
+  - I(config_file) should contain Tower configuration in the following format
+      host=hostname
+      username=username
+      password=password
+'''
+
+
+EXAMPLES = '''
+    tower_credential:
```
Comment (gundalow) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py]: Please add a `- name: ` and left align the examples
```
- name: Add tower credentials
  tower_credential:
    name: Team Name
```

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_user.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    defaults to find your Tower host information.
+  - I(config_file) should contain Tower configuration in the following format
+      host=hostname
+      username=username
+      password=password
+'''
+
+
+EXAMPLES = '''
+    tower_user:
```
Comment (gundalow) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_user.py]: Update example

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+            become_username = dict(),
+            become_password = dict(no_log=True),
+            vault_password = dict(no_log=True),
+            description = dict(),
+            organization = dict(required=True),
+            project = dict(),
+            tower_host = dict(),
+            tower_username = dict(),
+            tower_password = dict(no_log=True),
+            tower_verify_ssl = dict(type='bool', default='yes'),
```
Comment (gundalow) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py]: `default=True` (as it's a bool, not a string)

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+        description:
+          - Password for your Tower instance.
+        required: False
+        default: null
+    tower_verify_ssl:
+        description:
+          - Dis/allow insecure connections to Tower. If C(no), SSL certificates will not be validated.
+            This should only be used on personally controlled sites using self-signed certificates.
+        required: False
+        default: 'yes'
```
Comment (gundalow) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py]: `default: True`

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py (hunk 3 above)
Comment (gundalow) [reply lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py]: `default=True` (as it's a bool, not a string)

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_user.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        description:
+          - Password for your Tower instance.
+        required: False
+        default: null
+    tower_verify_ssl:
+        description:
+          - Dis/allow insecure connections to Tower. If C(no), SSL certificates will not be validated.
+            This should only be used on personally controlled sites using self-signed certificates.
+        required: False
+        default: 'yes'
```
Comment (gundalow) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_user.py]: `default: True`

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_user.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+            last_name = dict(),
+            password = dict(no_log=True),
+            email = dict(required=True),
+            organization = dict(),
+            superuser = dict(type='bool', default=False),
+            auditor = dict(type='bool', default=False),
+            tower_host = dict(),
+            tower_username = dict(),
+            tower_password = dict(no_log=True),
+            tower_verify_ssl = dict(type='bool', default='yes'),
```
Comment (gundalow) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_user.py]: `default=True` (as it's a bool, not a string)

**Review by gundalow — CHANGES_REQUESTED** [review]

**Review by gundalow — APPROVED** [review]

**Comment by matburt** [comment]
shipit

**Comment by ghjm** [comment]
shipit

**Pull request merged**